test/firewall_stage1_finish_test.rb

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env rspec
+
+require_relative "test_helper"
+
+require "yast"
+require_relative "../src/clients/firewall_stage1_finish"
+
+describe Yast::FirewallStage1FinishClient do
+  describe "main" do
+    context "when the client is called with 'Info'" do
+      it "returns a hash with 'steps', 'title' and 'when' keys" do
+
+        res = Yast::WFM.CallFunction("firewall_stage1_finish", ["Info"])
+
+        expect(res).to be_a(Hash)
+        expect(res).to include("steps", "title", "when")
+      end
+    end
+
+    context "when the client is called with 'Write'" do
+      before do
+        allow(Yast::WFM).to receive(:Args).and_return("Write")
+      end
+
+      context "in autoninst Mode" do
+        let(:installed) { false }
+
+        before do
+          allow(Yast::Mode).to receive(:autoinst).and_return(true)
+          allow(Yast::SuSEFirewall).to receive(:WriteConfiguration)
+          allow(Yast::Service).to receive(:Enable)
+          allow(Yast::SuSEFirewall4Network).to receive(:IsInstalled).and_return(installed)
+          allow(subject).to receive(:adjust_ay_configuration)
+        end
+
+        context "with SuSEfirewall2 installed" do
+          let(:installed) { true }
+
+          it "allows the remote installations in use in the SuSEFirewall configuration" do
```

Comment:
np: The description is confusing for me (maybe it is just me). What about *adjust the SuSEFirewall configuration*?